packages/beacon-node/src/monitoring/system.ts

```diff
@@ -5,264 +5,211 @@ import {Logger} from "@lodestar/utils";
-  return memoryNodeBytesTotal;
-}
+  get memoryNodeBytesTotal(): number {
+    return this._memoryNodeBytesTotal;
+  }
 
-/**
- * Amount of free memory in bytes
- */
-export function getMemoryNodeBytesFree(): number {
-  return memoryNodeBytesFree;
-}
+  get memoryNodeBytesFree(): number {
+    return this._memoryNodeBytesFree;
+  }
 
-/**
- * Amount of memory in bytes used by cache
- */
-export function getMemoryNodeBytesCached(): number {
-  return memoryNodeBytesCached;
-}
+  get memoryNodeBytesCached(): number {
+    return this._memoryNodeBytesCached;
+  }
 
-/**
- * Amount of memory in bytes used by buffers
- */
-export function getMemoryNodeBytesBuffers(): number {
-  return memoryNodeBytesBuffers;
-}
+  get memoryNodeBytesBuffers(): number {
+    return this._memoryNodeBytesBuffers;
+  }
 
-/**
- * Total amount of available disk space in bytes
- */
-export function getDiskNodeBytesTotal(): number {
-  return diskNodeBytesTotal;
-}
+  get diskNodeBytesTotal(): number {
+    return this._diskNodeBytesTotal;
+  }
 
-/**
- * Amount of free disk space in bytes
- */
-export function getDiskNodeBytesFree(): number {
-  return diskNodeBytesFree;
-}
+  get diskNodeBytesFree(): number {
+    return this._diskNodeBytesFree;
+  }
 
-/**
- * Total time spent in seconds on disk I/O operations
- */
-export function getDiskNodeIOSeconds(): number {
-  // Note: currently unused by beaconcha.in
-  return 0;
-}
+  get diskNodeIOSeconds(): number {
+    // Note: currently unused by beaconcha.in
+    return 0;
+  }
 
-/**
- * Total number of disk read I/O operations
- */
-export function getDiskNodeReadsTotal(): number {
-  return diskNodeReadsTotal;
-}
+  get diskNodeReadsTotal(): number {
+    return this._diskNodeReadsTotal;
+  }
 
-/**
- * Total number of disk write I/O operations
- */
-export function getDiskNodeWritesTotal(): number {
-  return diskNodeWritesTotal;
-}
+  get diskNodeWritesTotal(): number {
+    return this._diskNodeWritesTotal;
+  }
 
-/**
- * Total amount of bytes received over the network
- */
-export function getNetworkNodeBytesTotalReceive(): number {
-  return networkNodeBytesTotalReceive;
-}
+  get networkNodeBytesTotalReceive(): number {
+    return this._networkNodeBytesTotalReceive;
+  }
 
-/**
- * Total amount of bytes transmitted over the network
- */
-export function getNetworkNodeBytesTotalTransmit(): number {
-  return networkNodeBytesTotalTransmit;
-}
+  get networkNodeBytesTotalTransmit(): number {
+    return this._networkNodeBytesTotalTransmit;
+  }
 
-/**
- * Unix timestamp in seconds of boot time
- */
-export function getMiscNodeBootTsSeconds(): number {
-  return miscNodeBootTsSeconds;
-}
+  get miscNodeBootTsSeconds(): number {
+    return this._miscNodeBootTsSeconds;
+  }
 
-/**
- * Operating system, can be one of: lin, win, mac, unk for unknown
- */
-export function getMiscOs(): MiscOs {
-  return miscOs;
+  get miscOs(): MiscOs {
+    return this._miscOs;
+  }
 }
+
+export default new System();
```

Comment:
IMO the best singleton pattern is to just initialize the class and export is as default, this means that the class is always initialized but I don't see a problem with that